Framework/Algorithms/src/FilterEvents.cpp

```diff
@@ -272,8 +273,15 @@ void FilterEvents::exec() {
   // Form the names of output workspaces
   std::vector<std::string> outputwsnames;
   std::map<int, DataObjects::EventWorkspace_sptr>::iterator miter;
+  Goniometer inputGonio = m_eventWS->run().getGoniometer();
   for (miter = m_outputWorkspacesMap.begin();
        miter != m_outputWorkspacesMap.end(); ++miter) {
+    try {
+      DataObjects::EventWorkspace_sptr ws_i = miter->second;
+      ws_i->mutableRun().setGoniometer(inputGonio, true);
```

Comment:
Correct me if I am wrong, but if I understand correctly this issue had its origin from an attempt to share the run (to avoid duplicating large logs). Now that we set a new goniometer here we need to call `mutableRun()`, which breaks sharing of the `Run` and thus the issue with large duplicate logs is back?